specification/under_development/Compilation_And_Targeting.md

```diff
@@ -0,0 +1,67 @@
+# Compilation Process and Toolchain
+
+QIR serves as an integrating layer on top of which a shared compilation stage is built to connect a variety of front- and backends. To accelerate progress, quantum programming frameworks often choose to expose tools and APIs to facilitate reseach and development efforts across the entire stack rather than just at the application level. Efforts to develop and evaluate the performance of different error correction schemes, for example, illustrate that compilation logic may initially be expressed as application code before it is automated and integrated into the compiler toolchain. 
+A high degree of flexibility and configurability is hence of great importance when conceptualizing different compilation stages and how front- and backends integrate with the QIR toolchain and ecosystem.
+At the same time, a clear understanding of the purpose and functionality of subsequent stages in the compilation process is absolutely vital to grow a robust and performant compiler infrastructure over time.
+
+In addition to the [design goals of QIR](https://github.com/qir-alliance/qir-spec/blob/main/Scope.md) itself, we hence define the following goals for the compiler infrastructure around it:
+
+- Unify the ecosystem when reasonable; <br/>
+  Certain design choices can easily and often unintentionally lead to a fragmentation of the ecosystem of quantum application, libraries, and compilation tools, making it harder to leverage existing work in a new context. For example, requiring application code to be expressed in terms of a backend specific set of quantum instructions impedes portability of that code and hampers the development of shared application libraries.
+
+- Allow to make use of backend specific instructions at the application level; <br /> 
+  Exposing access to backend specific instructions is a valuable tool for research purposes. While we encourage libraries to be built against a hardware agnostic default set of instructions ([Default QIS]()), we aim for a simple mechanism for deviating from that default for the purpose of experimenting with new hardware intrinsics and developing backend specific compiler optimizations. These development in turn inform what abstractions are beneficial to expose in the Default QIS, and permit to ultimately incorporate fruitful optimizations into a target specific compilation stage.
+  
+- Define (and follow clear) guideance for defining backend specific instruction sets; <br />
+  ...
+
+TODO: the same as what is mentioned under bullet 1 applies to profiles
+
+## Compilation Stages
+
+- Language specific compilation: Compilation of source code to QIR bitcode
+- General optimizations: Optimizations that map QIR -> QIR and that do not limit compatibility in any way
+- General linking: Combines bitcode from different sources and libraries according to [standard linking practices]()
```

Comment:
Link missing? I suppose you mean like  [LLVM link](https://llvm.org/docs/CommandGuide/llvm-link.html)

Reply:
updated.